content/guides/development/rails/migration-guides/primer-dialog.mdx

```diff
@@ -0,0 +1,161 @@
+---
+title: Moving Away From <details-dialog>
+---
+
+This guide will show you how to upgrade from the now deprecated `<details-dialog>` element to the latest [`Primer::Alpha::Dialog`](/components/dialog/rails/alpha/)
+component.
+
+## Migrating from `<details-dialog>` to `Primer::Alpha::Dialog`
+`details-dialog` uses the `details` and `summary` elements to render a dialog. This is different than the [Primer `Dialog` component](https://primer.style/design/components/dialog/rails/alpha) that uses a container with `role="dialog"` attached.
+
+The `<details-dialog>` differs from the `Primer::Alpha::Dialog` in 3 different ways:
+
+1. The trigger uses the `summary` element. 
+2. The dialog's body (`details-dialog`) lives within the `details` element.
+3. The `<details>` wrapper contains both the trigger button (`summary`) and the dialog itself (`details-dialog`). 
+
+For example, here's how `details-dialog` might be used:
+
+```html
+<details>
+  <summary>Open dialog</summary>
+  <details-dialog>
+    Modal content
+    <button type="button" data-close-dialog>Close</button>
+  </details-dialog>
+</details>
+```
+
+Here's an example on how `Primer::Alpha::Dialog` might be used:
+
+```ruby
+render(Primer::Alpha::Dialog.new(title: 'Dialog Title', subtitle: 'Optional subtitle', size: :large)) do |d|
+  d.with_show_button { "Open dialog" }
+  d.with_body { "Modal content" }
+end
+```
+
+Both `Primer::Alpha::Dialog` and `<details-dialog>` will render similar dialogs, albeit with a few key differences:
+
+* Styling
+* Functionality and semantics
+
+### Styling differences
+`<details-dialog>` and `Primer::Alpha::Dialog` will have some minor style differences, such as with padding and margins that are applied by default.  
```

Comment:
```suggestion
`<details-dialog>` and `Primer::Alpha::Dialog` will have some minor style differences, such as with padding and margins that are applied by default. Please utilize these updated styles and refrain from overriding styles to make it appear like the previous design.
```

Just thinking we might want to make it clear that this is the new design and to please use it 😄 feel free to reword that